src/main.ts

```diff
@@ -13,6 +13,39 @@ async function setup(): Promise<void> {
   await setOutputs(name, url)
 }
 
+async function prolong(): Promise<void> {
+  const user: string = core.getInput('user')
+  const password: string = core.getInput('password')
+  const url: string = core.getInput('url')
+  const name: string = core.getInput('name')
+
+  const name_regex = new RegExp(
+    '(^[^/]+:)staging(-[0-9a-f]+.[^./]+.[0-9]+.[0-9]+)$'
+  )
+  const match = name.match(name_regex)
+  if (match == null) {
+    throw Error('The name is not one of Scality actions artifacts')
+  }
+  const artifacts_target = `${match[1]}prolonged${match[2]}`
+
+  const final_url: string = new URL(
+    path.join('/copy/', name, artifacts_target),
+    url
+  ).toString().concat("/")
+  const request_config: AxiosRequestConfig = {
+    auth: {
+      username: user,
+      password
+    },
+    maxBodyLength: Infinity,
+    maxContentLength: Infinity
+  }
+  await axios.get(final_url, request_config)
+  core.info(`copying '${name} to '${artifacts_target}`)
+  await setOutputs('name', artifacts_target)
```

Comment:
Check how this function is used, it's not the same as core.setOutputs.

Here you can call it once and set the name and the url on it.